ext/pdo_pgsql/pgsql_driver.c

```diff
@@ -1224,6 +1242,34 @@ PHP_METHOD(PDO_PGSql_Ext, pgsqlGetPid)
 }
 /* }}} */
 
+/* {{{ proto bool PDO::pgsqlSetNoticeCallback(mixed callback)
+   Sets a callback to receive DB notices (after client_min_messages has been set) */
+PHP_METHOD(PDO_PGSql_Ext, pgsqlSetNoticeCallback)
+{
+	pdo_dbh_t *dbh;
+	pdo_pgsql_db_handle *H;
+	zend_fcall_info fci = empty_fcall_info;
+	zend_fcall_info_cache fcc = empty_fcall_info_cache;
+	
+	dbh = Z_PDO_DBH_P(getThis());
```

Comment:
Because this is a method, `ZEND_THIS` can be used instead of `getThis()`. The difference between the two is that the former can only be used safely instead methods, while the latter can be used in procedural/method hybrids.

```suggestion
	dbh = Z_PDO_DBH_P(ZEND_THIS);
```